File: Data Structures & Algorithms/subarray-sum-equals-k/submission-4.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        mapping = {0: 1}
        currsum = 0
        res = 0

        for num in nums:
            currsum += num
            calc = currsum - k

            if calc in mapping:
                res += mapping[calc]

            mapping[currsum] = mapping.get(currsum, 0) + 1

        return res

File: Data Structures & Algorithms/concatenation-of-array/submission-4.java
class Solution {
    public int[] getConcatenation(int[] nums) {
        int n = nums.length; 
        int[] res = new int[n * 2];
        int idx = 0; 
        for(int i = 0; i < 2; i++){
            for(int j: nums){
                res[idx] = j; 
                idx += 1; 
            }
        }
        return res; 
    }
}